### PR TITLE
TS-13 Clock

### PR DESCRIPTION
**Pull request by Dreamless**

### CODE REVIEW

File: typescript/13-clock/clock.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export class Clock {
+  private readonly hour: number;
+  private readonly minute: number;
+
+  constructor(hour: number, minute: number = 0) {
+    const minInDay = 1440;
+    const totalMinutes = (((hour * 60 + minute) % minInDay) + minInDay) % minInDay;
```
Comment (frozenspider) [inline typescript/13-clock/clock.ts:7]: What are you achieving by doing `% minInDay` twice?

Comment (Dreamless) [reply typescript/13-clock/clock.ts]: Fixed: Separated logic for positive and negative time values.

Comment (Dreamless) [reply typescript/13-clock/clock.ts]: First minInDay brought the positive time value into the desired format, the second negative

Comment (frozenspider) [reply typescript/13-clock/clock.ts]: The initial approach was the correct one, I asked for the explanation and reasoning behind that logic. Do not bruteforce solutions.

Comment (Dreamless) [reply typescript/13-clock/clock.ts:7]: Returned to the original approach

Comment (frozenspider) [reply typescript/13-clock/clock.ts:7]: I still don't see a proper justification

Comment (Dreamless) [reply typescript/13-clock/clock.ts:7]: % minInDay results in the correct format of minutes even if the number is negative or greater than the number of days. If the time value is negative, we first convert it to positive by adding minInDay, and then convert it to the total number of minutes.

Comment (frozenspider) [reply typescript/13-clock/clock.ts:7]: What does "correct" format mean?

Comment (Dreamless) [reply typescript/13-clock/clock.ts:7]: ranging from 0 to 1440 which is equivalent to a day in minutes

**Review by frozenspider — APPROVED** [review]

**Pull request merged**